core/src/org/sbml/jsbml/AbstractSBase.java

```diff
@@ -1835,6 +1835,15 @@ public int hashCode() {
     return hashCode + prime * getLevelAndVersion().hashCode();
   }
 
+  /*
+   * (non-Javadoc)
+   * @see org.sbml.jsbml.SBase#hasExtension(java.lang.String)
+   */
+  @Override
+  public boolean hasExtension(String extName) {
+    return getModel().getExtension(extName) != null;
```

Comment:
Sometimes, an extension can be directly attached to the object itself, such as the extended species in FBC that has additional attributes. My suggestion would be that the method on `AbstractSBase` only checks if `this` object has any extension.